controllers/reconcilers/grafana/deployment_reconciler.go

```diff
@@ -211,6 +220,41 @@ func getContainers(cr *v1beta1.Grafana, scheme *runtime.Scheme, vars *v1beta1.Op
 	return containers
 }
 
+// getGrafanaContainerSecurityContext provides default securityContext for grafana container
+func getGrafanaContainerSecurityContext() *v1.SecurityContext {
+	capability := &v1.Capabilities{
+		Drop: []v1.Capability{
+			"ALL",
+		},
+	}
+	return &v1.SecurityContext{
+		AllowPrivilegeEscalation: model.BoolPtr(false),
+		ReadOnlyRootFilesystem:   model.BoolPtr(true),
+		Privileged:               model.BoolPtr(false),
+		RunAsNonRoot:             model.BoolPtr(true),
+		RunAsUser:                model.IntPtr(10001),
+		RunAsGroup:               model.IntPtr(10001),
+		Capabilities:             capability,
+	}
+}
+
+func getReadinessProbe(cr *v1beta1.Grafana) *v1.Probe {
+	return &v1.Probe{
+		ProbeHandler: v1.ProbeHandler{
+			HTTPGet: &v1.HTTPGetAction{
+				Path:   GrafanaHealthEndpoint,
+				Port:   intstr.FromInt(GetGrafanaPort(cr)),
+				Scheme: v1.URISchemeHTTP,
```

Comment:
Since we provide the entire readinessProbe object to the user they can easily overwrite the default http setting. But in general readinessProbe will use `http` and not `https`.

Reply:
In v4, we do not allow users to specify scheme, it's chosen automatically based on the server.protocol setting. There, with the old approach of being able to specify all fields, we also had a CRD-based enforcement for the list of server protocols (only http and https).
I guess there are not many people who would run grafana with HTTPs backend, I would rather expect a transparent case of service mesh.

Not saying we need to change anything here, just adding the comment for the context.

Reply:
Agreed, I think the limited amount of people using service mesh don't justify the reason to add any logic for this.
Thanks for the extra context.